File: Structural/Bridge/JavaScript/radio.js
// radio.js

const Device = require("./device");

class Radio extends Device {
  constructor() {
    super();
    this.on = false;
    this.volume = 30;
  }

  turnOn() {
    this.on = true;
    console.log("Radio is now ON.");
  }

  turnOff() {
    this.on = false;
    console.log("Radio is now OFF.");
  }

  setVolume(volume) {
    this.volume = volume;
    console.log(`Radio volume set to ${volume}`);
  }

  getVolume() {
    return this.volume;
  }

  isOn() {
    return this.on;
  }
}

module.exports = Radio;

File: Structural/Bridge/Python/radio.py
# radio.py

from device import Device

class Radio(Device):
  def __init__(self):
    self._on = False
    self._volume = 30

  def turn_on(self):
    self._on = True
    print("Radio is now ON.")

  def turn_off(self):
    self._on = False
    print("Radio is now OFF.")

  def set_volume(self, volume: int):
    self._volume = volume
    print(f"Radio volume set to {volume}")

  def get_volume(self) -> int:
    return self._volume

  def is_on(self) -> bool:
    return self._on
